lua/lint.lua

```diff
@@ -140,6 +143,16 @@ function M.try_lint(names, opts)
     linter.name = linter.name or name
     return linter
   end
+
+  -- kill any running process for this buffer
+  local bufnr = api.nvim_get_current_buf()
+  local previous_handles = buffer_to_running_handles[bufnr] or {}
+  for _, handle in ipairs(previous_handles) do
+    if handle and not handle:is_closing() then
+      handle:close()
+    end
```

Comment:
I'm not sure if calling `:close` is the right thing to do here as this can cause defunct processes. See https://github.com/libuv/libuv/issues/1911 

This would probably need to send a term signal instead.